website/next.config.mjs

```diff
@@ -0,0 +1,102 @@
+import nextra from 'nextra'
+import * as fs from 'fs';
+
+const reSourcePluginVersion = /VERSION_SOURCE_([a-zA-Z0-9_]+)/;
+const reDestPluginVersion = /VERSION_DESTINATION_([a-zA-Z0-9_]+)/;
+const reCLI = "VERSION_CLI";
+
+function getVersions() {
+  let versions = {
+    sources: {},
+    destinations: {},
+    cli: "",
+  }
+  const dir = fs.opendirSync('./versions')
+  let dirent
+  while ((dirent = dir.readSync()) !== null) {
+    if (dirent.isFile() && dirent.name.startsWith('source-')) {
+      let name = dirent.name.split('-')[1].split('.')[0]
+      let {latest} = JSON.parse(fs.readFileSync('./versions/' + dirent.name, 'utf8'))
+      versions.sources[name] = latest.split('-')[3]
+    } else if (dirent.isFile() && dirent.name.startsWith('destination-')) { 
+      let name = dirent.name.split('-')[1].split('.')[0]
+      let {latest} = JSON.parse(fs.readFileSync('./versions/' + dirent.name, 'utf8'))
+      versions.destinations[name] = latest.split('-')[3]
+    } else if (dirent.isFile() && dirent.name == "cli.json") {
+      versions.cli = JSON.parse(fs.readFileSync('./versions/' + dirent.name, 'utf8')).latest.split('-')[1]
+    }
+  }
+  dir.closeSync()
+  return versions
+}
```

Comment:
```suggestion
function getVersions() {
  const files = fs
    .readdirSync("./versions", { withFileTypes: true })
    .filter((dirent) => dirent.isFile())
    .map((file) => ({
      name: path.basename(file.name, path.extname(file.name)),
      latest: JSON.parse(fs.readFileSync(`./versions/${file.name}`, "utf8"))
        .latest,
    }));

  const cli = files.find((file) => file.name === "cli").latest.split("-")[1];
  const sources = Object.fromEntries(
    files
      .filter((file) => file.name.startsWith("source-"))
      .map((file) => [file.name.split("-")[1], file.latest.split("-")[3]])
  );
  const destinations = Object.fromEntries(
    files
      .filter((file) => file.name.startsWith("destination-"))
      .map((file) => [file.name.split("-")[1], file.latest.split("-")[3]])
  );

  return { cli, sources, destinations };
}
```

[Non blocking] I think using `readdirSync` instead of a `while` loop makes this code easier to read

Reply:
let's do it in a follow-up PR as already have a fatigue from this one and hard to test various changes